File: mobile/lib/features/b2b/domain/entities/payment_status.dart
enum PaymentStatus {
  pending,
  partial,
  paid,
  credit,
  overdue;

  String get displayName {
    switch (this) {
      case PaymentStatus.pending:
        return 'En attente';
      case PaymentStatus.partial:
        return 'Partiel';
      case PaymentStatus.paid:
        return 'Payé';
      case PaymentStatus.credit:
        return 'À crédit';
      case PaymentStatus.overdue:
        return 'En retard';
    }
  }

  static PaymentStatus fromString(String value) {
    return PaymentStatus.values.firstWhere(
      (e) => e.name.toUpperCase() == value.toUpperCase(),
      orElse: () => PaymentStatus.pending,
    );
  }
}

File: backend/jufa-backend/src/main/java/ml/jufa/backend/b2b/entity/PaymentStatus.java
package ml.jufa.backend.b2b.entity;

public enum PaymentStatus {
    PENDING("En attente"),
    PARTIAL("Partiel"),
    PAID("Payé"),
    CREDIT("À crédit"),
    OVERDUE("En retard");

    private final String displayName;

    PaymentStatus(String displayName) {
        this.displayName = displayName;
    }

    public String getDisplayName() {
        return displayName;
    }
}
